Docs, Starcraft 2: Add French documentation for setup and game page

neocerber opened the pull request:
## What is this fixing or adding?
- Add a French version of the setup and game page for Starcraft 2.
  - Except for the "Progression Balancing" section, the content of the French setup/game page should be a direct translation of its English counter-part.
- Adapt the Starcraft 2 ?world class? for multiple tutorials.
- Add a link to the French game page in the English game page to make potential French user aware that it exist (Without that, the French game page is only available by changing the url by hand.)
- Corrected the English game page first header (Starcraft 2 is no longer limited to Wing of Liberty) and a sentence.


## How was this tested?
By locally hosting the website, the rendering of the French setup and game page was confirmed. It was also confirmed that the English setup and game page remains correctly rendered. All the url and link of the French setup and game page were confirmed to redirect at the correct page/location.

## If this makes graphical changes, please attach screenshots.
None that are relevant.

Ziktofel commented:
I don't speak French but the French manual appears to have all the sections as the English one does.

benny-dreamly commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+- [*StarCraft 2*](https://starcraft2.com/en-us/)
+- [La version la plus récente d'Archipelago](https://github.com/ArchipelagoMW/Archipelago/releases)
+
+## Comment est-ce que j'installe ce *randomizer*?
```
randomizer ou randomiseur?

benny-dreamly replied:
Je sais que vous avez mis cela en gras alors ce n'est pas important, mais pour un peu de "consistency".

neocerber replied:
I like the idea of keeping the English word since I feel that this terms has more meaning in our context and that translating it would be akin to translate "Starcraft 2". Is that acceptable?

benny-dreamly replied:
Yep!

benny-dreamly commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+## Où est-ce que j'obtiens le fichier de configuration (i.e., le *yaml*) pour ce jeu?
+
+Un fichier dans le format *yaml* est utilisé pour communiquer à Archipelago comment vous voulez que votre jeu soit 
+*randomized*. 
```
randomisé?

neocerber replied:
I like the idea of keeping the English word since I feel that this terms has more meaning in our context and that translating it would be akin to translate "Starcraft 2". Is that acceptable?

benny-dreamly replied:
Yeah perfectly acceptable in this case.

benny-dreamly commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+   - Uniquement pour cette étape, les utilisateurs de macOS devraient plutôt suivre les instructions qui se trouvent à 
+["Exécuter sous macOS"](#exécuter-sous-macos).
+3. Dans le client de *StarCraft 2 Archipelago*, écrire la commande `/download_data`. Cette commande va lancer 
+l'installation des *Maps* et des *Data files* qui sont nécessaires pour jouer à *StarCraft 2 Archipelago*.
```
cartes? (cela n'est pas important, juste des suggestions)

neocerber replied:
I decided to side-step the problem by replacing both english terms by "fichiers". Good with you?

benny-dreamly replied:
To be perfectly honest it sounds fine!

benny-dreamly replied:
Honestly it's also fairly general and understandable too so that makes lots of sense.

agilbert1412 replied:
Je suis en désaccord. Le dossier sur le disque s'appellera toujours "Maps", donc l'appeller comme tel rend ça plus clair si la personne a besoi de vérifier si ça  a marché, le path, etc

benny-dreamly commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+Si vous désirez des informations et/ou instructions générales sur l'utilisation d'un fichier *yaml* pour Archipelago, 
+veuilliez consutler [*Creating a YAML*](/tutorial/Archipelago/setup/en#creating-a-yaml).
+
+### Questions récurrentes à propos du fichier yaml
```
yaml en plus de formatting? (only for consistency)

benny-dreamly replied:
the others are italicized (I gave up on typing in French as it's too early for my brain atm)

neocerber replied:
True!

benny-dreamly commented on worlds/sc2/docs/setup_en.md:
```diff
@@ -88,7 +88,7 @@ specific description of what's going wrong and attach your log file to your mess
 ## Running in macOS
 
-To run StarCraft 2 through Archipelago in macOS, you will need to run the client via source as seen here: [macOS Guide](https://archipelago.gg/tutorial/Archipelago/mac/en). Note: when running the client, you will need to run the command `python3 Starcraft2Client.py`.
+To run StarCraft 2 through Archipelago in macOS, you will need to run the client via source as seen here: [macOS Guide](https://archipelago.gg/tutorial/Archipelago/mac/en). Note: to lauch the client, you will need to run the command `python3 Starcraft2Client.py`.
```
launch*

neocerber replied:
Good catch!

benny-dreamly reviewed (commented):
overall looks pretty good I gave up suggesting things as it's too early for my French brain to function. I only looked at specifically the setup guide and such (I don't play Starcraft so I can't give feedback on the specifics but overall it looks fairly good)

neocerber commented:
Thank you benny-dreamly for the comments! I applied some of them but I would require your input on those kept open.

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+
+## Quel est l'effet de la randomisation sur ce jeu ?
+
+Les éléments qui suivent sont les *items* qui sont randomisés et qui doivent être débloqués pour être utilisé dans le 
```
utilisés*

MaxTousss replied:
True!

neocerber replied:
Done!

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+
+Les éléments qui suivent sont les *items* qui sont randomisés et qui doivent être débloqués pour être utilisé dans le 
+jeu:
+1. La capacité de produire des unités, outre les drones/probes/scv.
```
outre -> excepté?

MaxTousss replied:
Changed

neocerber replied:
Alright!

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+1. La capacité de produire des unités, outre les drones/probes/scv.
+2. Des améliorations spécifiques à certaines unités incluant quelques combinaisons qui ne sont pas disponibles dans les 
+campagnes génériques, comme le fait d'avoir les deux types d'évolution en même temps pour une unité *Zerg* et toutes 
+les améliorations de la Lance d'Adun simultanément pour les *Protoss*.
```
Spear of Adun est un nom propre selon moi

neocerber replied:
True. Also changed Hyperion for that same reason. Kept "Léviathan", except changing "le" for "un", since it is one among many, in theory. Is that acceptable?

agilbert1412 replied:
Fine by me, especially since Leviathan's translation is pretty much the exact same word. No one will get confused. Spear on the other hand would be more annoying I think

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+3. L'accès aux améliorations génériques des unités, e.g. les améliorations d'attaque et d'armure.
+4. D'autres améliorations diverses telles que les améliorations de laboratoire et les mercenaires pour les *Terran*, les 
+niveaux et les améliorations de Kerrigan pour les *Zerg*, et les améliorations de la Lance d'Adun pour les *Protoss*.
+5. Avoir des *minerals*, du *vespene gaz*, et du *supply* au début de chaque mission.
```
gas* je crois

neocerber replied:
You are right!

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+5. Avoir des *minerals*, du *vespene gaz*, et du *supply* au début de chaque mission.
+
+Les *items* sont trouvés en accomplissant du progrès dans les catégories suivantes:
+* Accomplir des missions
```
Accomplir -> Completer

neocerber replied:
Completer seems to be discouraged in this context. If the word "Accomplir" seems incorrect, what about "terminer"?

agilbert1412 replied:
How is it discourage?
I think terminer sounds a bit like "you don't have to win, just finish", kind of like the "win" condition on In Utter Darkness.
Still, I'd accept it for missions, but it really doesn't fit for bonus objectives, which then would be better with "Accomplir", although I still prefer "Completer". Either way, not attached to this at all, you can do what you prefer.

neocerber replied:
Using "Compléter" as to finish is an anglicism from "to complete", it seems: https://vitrinelinguistique.oqlf.gouv.qc.ca/21486/les-emprunts-a-langlais/emprunts-semantiques/emploi-deconseille-de-lemprunt-completer

I kept "terminer" for mission and switched to "Réussir" for "objectifs" and "défis", since it felt nice.

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+
+Les *items* sont trouvés en accomplissant du progrès dans les catégories suivantes:
+* Accomplir des missions
+* Accomplir les objectifs supplémentaires (e.g., récolter le matériel pour les recherches dans *Wings of Liverty*)
```
Liberty*

neocerber replied:
Ouuff, thanks for catching that XD

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+
+## Quelles sont les modifications non aléatoires comparativement à la version de base de *Starcraft 2*
+
+1. Certaines des missions ont plus de *vespene geyses* pour permettre l'utilisation d'une plus grande variété d'unités.
```
geysers*

neocerber replied:
Applied!

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+2. Plusieurs unités et améliorations ont été ajoutées sous la forme d*items*.
+Ils proviennent de la version *co-op*, *melee*, des autres campagnes, d'expansions ultérieures, de *Brood War*, ou de 
+l'imagination des développeurs de *StarCraft 2 Archipelago*.
+3. Les structures de production, e.g. *Factory*, *Starport*, *Robotics Facility*, and *Stargate*, n'ont plus de requis 
```
requis -> exigence

neocerber replied:
Done!

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -0,0 +1,87 @@
+Consulter [*Advanced YAML Guide*](https://archipelago.gg/tutorial/Archipelago/advanced_settings/en) pour savoir comment 
+changer ça.
+
+## Commandes du client qui sont uniques à ce jeux
```
ce jeu*

neocerber replied:
True!

agilbert1412 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+Utilisez la page [*Weighted Options*](/weighted-options) pour avoir accès à ces dernières.
+
+Si vous désirez des informations et/ou instructions générales sur l'utilisation d'un fichier *yaml* pour Archipelago, 
+veuilliez consutler [*Creating a YAML*](/tutorial/Archipelago/setup/en#creating-a-yaml).
```
veuillez*

neocerber replied:
Done!

agilbert1412 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+veuilliez consutler [*Creating a YAML*](/tutorial/Archipelago/setup/en#creating-a-yaml).
+
+### Questions récurrentes à propos du fichier yaml
+#### Comment est-ce que je sais que mon *yaml* est bien défini?
```
Rien sur le fait de le tester sur le site? il y a même une page exprès pour https://archipelago.gg/check

neocerber replied:
TIL! I didn't it even existed!

The paragraph was changed to add this information, both in the French and English documentation.

agilbert1412 replied:
I'll take a look!

agilbert1412 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+Pour cette raison, cet ajustement a souvent peu d'influence sur la capacité de progresser dans un monde de 
+*StarCraft 2*. 
+Notez que l'utilisation de cette option, i.e. au moins un monde avec une valeur différente de *Progression Balancing*, 
+augmente le temps de génération d'un *MultiWorld*, alors certains recommandent de ne pas toucher à cette option.
```
certains recommendent de la laisser à zéro*

neocerber replied:
The default is 50. So, you suggest that users of SC2 use 0 which, if memory serves right, simply deactivate balancing for that word? 
If yes, I propose "certains recommendent de le définir à zéro."

agilbert1412 replied:
Well, using the default of 50 doesn't fit the rest of the sentence. 50 Will increase the generation time, as will any non-zero value, as it adds a step during generation.

So either the remainder of the sentence should be adjusted to properly explain that leaving it default is "fine", without claiming it is faster, or the recommendation should be zero, as zero is the fast value. Either is fine in my opinion, as the difference will not be noticeable to average players.

neocerber replied:
Modified so that recommendation is zero.

agilbert1412 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,191 @@
+#### Comment est-ce que je fais pour connaître le nom des *items* dans *StarCraft 2 Archipelago*? 
+
+Vous trouverez la liste complète des *items* de *StarCraft 2 Archipelago* à la page 
+[*Icon Repository*](https://matthewmarinets.github.io/ap_sc2_icons/).
```
pas de mention du data package?

neocerber replied:
Hmm, well, I have do idea what that is and it is note in the English version. Do you have a ref? I will read on it and incorporate the info., if I can, in both language.

neocerber replied:
Data package was mentioned. Since it also provides locations names, changed the title to include location. Since location is in the title, a mention that they are easy to find in the client was also made.

This was done in the French and English documention for uniformity.

agilbert1412 reviewed (changes requested):
Found a few typos, had a few suggestions which I am not particularly attached to, feel free to ignore

I did not review the MacOS guide

Bicoloursnake reviewed (approved):
I can't read French, but the correction to my sloppy macOS section on the English doc is appreciated. The tutorial object appears to be constructed correctly as well, though I'm away from my machine where I'd normally test to confirm that everything loads correctly on the webhost.

neocerber commented:
Thank you @agilbert1412  for the comments! I applied some of them but I would like your input on those kept open.

agilbert1412 commented on worlds/sc2/docs/fr_Starcraft 2.md:
```diff
@@ -40,11 +42,11 @@ Les choix faits dans le fichier *yaml* définissent la disposition des missions
 Ils proviennent de la version *co-op*, *melee*, des autres campagnes, d'expansions ultérieures, de *Brood War*, ou de 
 l'imagination des développeurs de *StarCraft 2 Archipelago*.
-3. Les structures de production, e.g. *Factory*, *Starport*, *Robotics Facility*, and *Stargate*, n'ont plus de requis 
+3. Les structures de production, e.g. *Factory*, *Starport*, *Robotics Facility*, and *Stargate*, n'ont plus d'exigence 
```
exigences* technologiques*, pluriel

neocerber replied:
Applied!

agilbert1412 reviewed (approved):
Still a few comments of mine left open, but they are all so minor that I feel like I can approve this.

neocerber commented:
All the suggestions of agilbert1412 were considered. I left two of them unresolved since they required larger changes that might be of interest to review again. However, agilbert1412 has already indicated that the current version seems acceptable.

Some other changes, that were not requested, were made:
- Changed all Starcraft to StarCraft for uniformity.
- Changed all link to Archipelago own pages to relative like so that it works even when self-hosting.
    - I validated all the link after the changes. 
- Information about the datapackage was also added in the English part for uniformity with the proposed French documentation.

MatthewMarinets commented on worlds/sc2/docs/setup_en.md:
```diff
@@ -66,9 +68,15 @@ start_inventory:
+
+You can also look up a complete list of the item names in the [Icon Repository](https://matthewmarinets.github.io/ap_sc2_icons/) page.
+This page also contains supplementary information of each item.
+However, the items shown in that page might differ from those shown in the datapackage page of Archipelago since the former is generated, most of the time, from the most recent developments of StarCraft 2 Archipelago.
```
"from the most recent developments of StarCraft 2 Archipelago" -> "from beta versions of StarCraft 2 Archipelago undergoing development."

neocerber replied:
Alright!

MatthewMarinets commented on worlds/sc2/docs/setup_en.md:
```diff
@@ -66,9 +68,15 @@ start_inventory:
-#### How do I know the exact names of items?
+#### How do I know the exact names of items and locations?
+
+The [*datapackage*](/datapackage) page of Archipelago website provides a complete list of the items and locations for each game that it currently supports, including StarCraft 2.
```
> page of *the* Archipelago website

Missing "the"

neocerber replied:
True!

MatthewMarinets reviewed (approved):
Only reviewed the English parts, and I have not yet run the Webhost locally to verify the links to french pages show up. I noticed some minor grammatical issues in the English docs, but otherwise looks solid.

neocerber commented:
The suggestions of MatthewMarinets are all resolved.

benny-dreamly reviewed (approved):
forgot to approve this will do that now

neocerber commented:
- [ ] Need to add a mention that the mod/game might break if not setted to english

neocerber commented:
- Added a mention that the mod/game might break if not setted to english
- Added instruction for custom hotkey
- Removed correction of game name in english part since proposed in #3074

Bicoloursnake commented:
English part still looks good after the recent commits.

Bicoloursnake commented:
I tested locally and can confirm that the relative links work as expected, and that the French documentation is accessible as expected (I still can't read most of the French though).

Thelk641 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,214 @@
+
+#### À quoi sert l'option *Progression Balancing*?
+
+Pour un monde *StarCraft 2* seule, cette option ne fait rien. 
```
This entire paragraph feels very "robotic". It should be something like : 

"Pour *Starcraft 2*, cette option ne fait pas grand-chose.
Il s'agit d'une option d'Archipelago permettant d'équilibrer la progression des mondes en interchangeant les *items* dans les *sphères*.  Cependant, *Starcraft 2* est beaucoup plus permissif en termes d'*items* qui permettent de progresser, ce réglage à donc peu d'influence sur la progression. Vu qu'il augmente le temps de génération d'un *MultiWorld*, nous recommandons de le désactiver pour *Starcraft 2*.

neocerber replied:
Applied except:
- Kept the sentence that describe what the option odes 
- Kept the fact that deactivation is by using the zero value
- Explicited that "sur la progression." with "sur la progression Starcraft 2."

neocerber replied:
The reviewer agreed after a discussion in private messages.

Thelk641 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,214 @@
+[*Icon Repository*](https://matthewmarinets.github.io/ap_sc2_icons/). 
+Notez que cette page contient diverses informations supplémentaires sur chacun des *items*.
+Cependant, l'information présente dans cette dernière peut différer de celle du *datapackage* d'Archipelago puisqu'elle 
+est générée, habituellement, à partir des développements récents de *StarCraft 2 Archipelago* qui n'ont peut-être pas 
```
"Développements récents" can be understood as "recent news" instead of "latest updates". Instead, it should be "à partir de la dernière version de *Starcraft 2 Archipelago*".

neocerber replied:
"développements récents" was meant as the dev branch. I feel that " dernière version de Starcraft 2 Archipelago" would mean the lastest version merged to Archipelago, which is not what it does. With that context, is that alright?

Thelk641 replied:
I guess the best way to say it would be to explicitely say that the version merged in Archipelago might not be the last version of the randomizer instead. So, instead something like "Cependant, il est possible que le site web d'Archipelago utilise une précédente version de *Starcraft 2 Archipelago*, auquel cas l'information présente sur cette image peut différer de celle de votre monde", or something like that ? Or we could call it a beta version, or a bleeding edge version, or something like that.

neocerber replied:
A modification, agreed in private messages, was applied.

Thelk641 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,214 @@
+4. Si le serveur a un mot de passe, l'inscrire lorsque vous y êtes invité.
+5. Une fois connecté, aller sur l'onglet *StarCraft 2 Launcher* dans le client. Dans cet onglet, vous devriez trouver 
+toutes les missions de votre monde. Les missions qui ne sont pas disponibles présentement auront leur texte dans une 
+tonte de gris. Vous n'avez qu'à cliquer une des missions qui est disponible pour la commencer!
```
Typo (or disagreement between different regions ?) : "tonte de gris" should be "nuance de gris"

neocerber replied:
Applied as suggested!

Thelk641 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,214 @@
+
+## Exécuter sous Linux
+
+Pour éxécuter *StarCraft 2* via Archipelago sous Linux, vous allez devoir installer le jeu avec *Wine* et ensuite 
```
Typo : "éxécuter" should be "exécuter", no accent on the first E. This typo is also present on lines 169, 175, 183, 187, 204, 205 and 210.

neocerber replied:
Done!

Thelk641 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,214 @@
+
+Pour que *StarCraft 2 Archipelago* utilise votre profil, suivez les étapes suivantes.
+Lancez *Starcraft 2* via l'application *Battle.net*. 
+Changez votre profil de raccourcis clavier pour le mode standard et acceptez, puis changez pour votre profil 
```
"puis changez pour votre profil personnalisé" works, but "puis sélectionnez votre profil personnalisé" vous work even better.

neocerber replied:
Applied as suggested!

Thelk641 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,214 @@
+Lancez *Starcraft 2* via l'application *Battle.net*. 
+Changez votre profil de raccourcis clavier pour le mode standard et acceptez, puis changez pour votre profil 
+personnalisé et acceptez. 
+Vous aurez juste besoin de faire ça une seule fois.
```
"Vous n'aurez besoin de faire ça qu'une seule fois" sounds a bit better, but this is a nitpick, both work.

neocerber replied:
Applied as suggested!

Thelk641 commented on worlds/sc2/docs/setup_fr.md:
```diff
@@ -0,0 +1,214 @@
+PATH_TO_ARCHIPELAGO=
+
+# Obtiens la dernière version de l'AppImage de Archipelago dans le dossier PATH_TO_ARCHIPELAGO.
+# Si PATH_TO_ARCHIPELAGO n'est pas déféni, la valeur par défaut est le dossier qui contient ce script.
```
Typo : déféni should be défini.

neocerber replied:
Done!

Thelk641 reviewed (commented):
Overall, it all sounds fine to me. One paragraph feels very robotics, but the rest is okay for me as a native French speaker from the old world.

neocerber commented:
All suggestions were processed. Every part of the French and English modifications were checked at least once. The validity of the links were tested by both the author and a peer.
